src/librustc_errors/emitter.rs

```diff
@@ -24,7 +24,23 @@ const ANONYMIZED_LINE_NUM: &str = "LL";
 /// Emitter trait for emitting errors.
 pub trait Emitter {
     /// Emit a structured diagnostic.
-    fn emit(&mut self, db: &DiagnosticBuilder<'_>);
+    fn emit_diagnostic(&mut self, db: &DiagnosticBuilder<'_>);
+
+    /// Emit a directive. The default is to do nothing; this should only be
+    /// produced with --error-format=json.
+    //
+    // njn: is "directive" a good name here? "json_directive"?
+    // "build_directive"?
```

Comment:
Anything derivative from the word "directive" seems reasonable to me! If you want to be super explicit it could be `maybe_emit_json_directive`